### ISSUE TITLE
Test errors for numpy 1.11 / pandas 0.18.0

### ISSUE THREAD

**Issue by matthew-brett** [pydata/pandas#12729]
See: https://travis-ci.org/matthew-brett/manylinux-testing/jobs/119188668#L3214

```
======================================================================
ERROR: test_range_slice_day (pandas.tseries.tests.test_period.TestPeriodIndex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/matthew-brett/manylinux-testing/venv/lib/python2.7/site-packages/pandas/tseries/tests/test_period.py", line 2603, in test_range_slice_day
    idx[v:]
  File "/home/travis/build/matthew-brett/manylinux-testing/venv/lib/python2.7/site-packages/pandas/tseries/base.py", line 175, in __getitem__
    result = getitem(key)
IndexError: failed to coerce slice entry of type str to integer
======================================================================
ERROR: test_range_slice_seconds (pandas.tseries.tests.test_period.TestPeriodIndex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/matthew-brett/manylinux-testing/venv/lib/python2.7/site-packages/pandas/tseries/tests/test_period.py", line 2660, in test_range_slice_seconds
    idx[v:]
  File "/home/travis/build/matthew-brett/manylinux-testing/venv/lib/python2.7/site-packages/pandas/tseries/base.py", line 175, in __getitem__
    result = getitem(key)
IndexError: failed to coerce slice entry of type str to integer
```

No such error testing against numpy 1.10.4 : https://travis-ci.org/matthew-brett/manylinux-testing/jobs/119039450#L523

**Comment by jreback** [comment]
This is at odds with the wheels built directly by the numpy on master. see [here](https://travis-ci.org/pydata/pandas/jobs/118605953). 2 known failures on pandas side ATM due to some compat issues. This is master numpy / master pandas.

The error you are seeing is rather old IIRC.

**Comment by jreback** [comment]
We have this changing from an ``IndexError`` in numpy < 1.11 to a ``TypeError`` in 1.11. Was this changed reverted before 1.11?
@charris @shoyer 

```
In [32]: arr = np.arange(0,2).astype('M8[ns]')

In [33]: arr
Out[33]: 
array(['1969-12-31T19:00:00.000000000-0500',
       '1969-12-31T19:00:00.000000001-0500'], dtype='datetime64[ns]')

In [34]: arr[slice('2014',None,None)]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-34-5c67b069ad87> in <module>()
----> 1 arr[slice('2014',None,None)]

IndexError: invalid slice

In [35]: np.__version__
Out[35]: '1.10.4'
```

```
>>> np.__version__
'1.11.0b3'
>>> arr = np.arange(0,2).astype('M8[ns]')
>>> arr
array(['1970-01-01T00:00:00.000000000', '1970-01-01T00:00:00.000000001'], dtype='datetime64[ns]')
>>> arr[slice('2014',None,None)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: failed to coerce slice entry of type str to integer
```

```
>>> np.__version__
'1.12.0.dev0+75c5af3'
>>> arr = np.arange(0,2).astype('M8[ns]')
>>> arr[slice('2014',None,None)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: slice indices must be integers or None or have an __index__ method
```

**Comment by charris** [comment]
I don't see where you tested with 1.11.0, the reversion was in 1.11.0rc1
```
commit 466d60d364dfc979ccffa58b5c8212137e6b2a27
Author: Charles Harris <charlesr.harris@gmail.com>
Date:   Sun Feb 14 11:49:29 2016 -0700

    Revert #6271 from charris/change-deprecated-indexes-to-error

    This reverts commit 0b0206c1617841ed2e5324de752ee8ede2cce791,
    reversing changes made to 438cdd3d75a0bb606e7ab7f96e59744c9b78d748.

    This reverts all the changes of indexing deprecation warnings to
    errors that were done prior to branching Numpy 1.11.x. It is easier
    to make a clean reversion here than to pick out those bits involving
    integer indexing specifically. As the aim of this reversion is to
    give downstream projects a bit more time to adapt I don't think the
    slight overkill is worth worrying about, especially if downstream
    projects start paying close attention to deprecations.

    The 1.11.0-notes also need updating, but I will submit a separate
    PR against master for that.

    Closes #7162.
```

**Comment by jreback** [comment]
it's the same (forgot to pull tags)

ok so you reverted this change - will update on our side
